api/proto/teleport/legacy/types/types.proto

```diff
@@ -6213,6 +6215,125 @@ enum PluginStatusCode {
   SLACK_NOT_IN_CHANNEL = 10;
 }
 
+// PluginOktaStatusDetailsV1 contains the details for the running Okta plugin.
+message PluginOktaStatusDetailsV1 {
+  // SSODetails are status details relating to SSO.
+  PluginOktaStatusDetailsSSO sso_details = 2;
+
+  // AppGroupSyncDetails are status details relating to synchronizing apps and
+  // groups from Okta.
+  PluginOktaStatusDetailsAppGroupSync app_group_sync_details = 3;
+
+  // UsersSyncDetails are status details relating to synchronizing users from
+  // Okta.
+  PluginOktaStatusDetailsUsersSync users_sync_details = 4;
+
+  // ScimDetails are status details relating to SCIM integration with
+  // Okta.
+  PluginOktaStatusDetailsSCIM scim_details = 5;
+
+  // AccessListSyncDetails are status details relating to synchronizing access
+  // lists from Okta.
+  PluginOktaStatusDetailsAccessListsSync access_lists_sync_details = 6;
+}
+
+// PluginOktaStatusDetailsSSO are details related to the
+// current status of the Okta integration w/r/t SSO.
+message PluginOktaStatusDetailsSSO {
+  bool enabled = 1;
+  string app_id = 2;
+  string app_name = 3;
+}
+
+// PluginOktaStatusDetailsAppGroupSync are details related to the
+// current status of the Okta integration w/r/t application and group
+// sync.
+message PluginOktaStatusDetailsAppGroupSync {
+  // LastSuccessful is the date of the last successful run.
+  google.protobuf.Timestamp last_successful = 1 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = true,
+    (gogoproto.jsontag) = "last_successful"
+  ];
+
+  // LastFailed is the date of the last failed run.
+  google.protobuf.Timestamp last_failed = 2 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = true,
+    (gogoproto.jsontag) = "last_failed"
+  ];
+
+  // NumAppsSynced is the total number of apps synchronized.
+  int32 num_apps_synced = 3;
+
+  // NumAppsSynced is the total number of groups synchronized.
+  int32 num_groups_synced = 4;
+}
+
+// PluginOktaStatusDetailsUsersSync are details related to the
+// current status of the Okta integration w/r/t users sync.
+message PluginOktaStatusDetailsUsersSync {
+  // Enabled is whether the users sync is enabled.
+  bool enabled = 1;
```

Comment:
Are we allowing users to control enabling or disabling detailed sync status reporting feature? My gut says we should have them enabled by default if the sync feature itself is enabled.

Reply:
This is a reflection of whether the service was enabled or not in the last status update interval, as the various sub-services can be independently switched on or off. This doesn't _control_ whether the sub-services run or not.

Reply:
Noted, thanks for clarifying. So are these status just to show up in a dashboard as a current state of sync config and status, and not to report a sync runner live? 

Pardon my lack of context here, but I am coming from following two assumptions:
- If the field does not control how the sync runs, it should not be in this struct. In the web ui, that can be fetched from the actual sync config store.
- If the service is switched off, the synchronization should not have happened in the first place. Meaning an individual sync run would never hold a falsy value?

Reply:
So, some background: 

The general idea was that we would pass this status struct to the UI unchanged, and it would be a bundle containing _everything_ the UI would need to know in one place. We didn't want the UI to necessarily have to know how to parse the whole plugin record.

We also considered just not creating sub-status struct values and sending `nil` for disabled services, rather than an `enabled: false` field in each sub-status block. In the end we decided that it was better to have a positively-specified value to distinguish it from "oops, we just forgot to send the status"; similar to having "This Page Intentionally Left Blank" on an exam paper or contract.

So, to your specific points:

1. Yes, but we specifically wanted to decouple the UI-visible status from the mechanisms that control the plugin. 
2. I'm not sure what you mean by _"an individual sync run would never hold a falsy value?"_. If you mean that it's essentially a static value, then yes: if it is `false` on one status update it will probably _always_ be false, as there will be no enabled service what would be updating it.

All of that said, these are the decisions we made during the hackathon; they are not necessarily the _best_ design decisions. If there are better patterns - or even more established patterns in Teleport - I'm totally open to a better design.

Reply:
Thank you again for more context. 
Sounds good to me if this was already discussed and agreed upon. It just looked like the fields would require data wrangling more than necessary.

Reply:
I think the addition of an overall status code for each service helps a bit (thanks for the suggestion). Having to compare timestamps to decide whether things are working was indeed a bit clunky. I still think its a useful diagnostic tool to know when the last failure was, though, so I have kept that timestamp in addition to adding the status code.